src/error.rs

```diff
@@ -1,39 +1,124 @@
 #![allow(overflowing_literals)]
 
+use crate::runtime::*;
+use crate::*;
+
+/// The ErrorCode (a.k.a HRESULT) of an error
+#[repr(transparent)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ErrorCode(pub i32);
+
+/// A WinRT related error
+pub struct Error {
+    code: ErrorCode,
+    info: IRestrictedErrorInfo,
+}
+
 /// An alias for `std::result::Result<T, winrt::Error>`
 #[must_use]
 pub type Result<T> = std::result::Result<T, Error>;
 
 impl<T> std::convert::From<Result<T>> for ErrorCode {
     fn from(result: Result<T>) -> Self {
         if let Err(error) = result {
-            // TODO: call SetErrorInfo
+            if !error.info.is_null() {
+                unsafe {
+                    SetRestrictedErrorInfo(error.info.as_raw() as _);
+                }
+            }
             return error.code();
         }
 
         ErrorCode(0)
     }
 }
 
-/// A WinRT related error
-#[derive(Debug)]
-pub struct Error {
-    code: ErrorCode,
-    // TODO: add `info: IErrorInfo`
+impl std::convert::From<ErrorCode> for Error {
+    fn from(code: ErrorCode) -> Self {
+        unsafe {
```

Comment:
I would really love for us to be in the habit of only marking the code that actually requires `unsafe` as such. Doing it like this makes it much harder to know what code needs to have closer inspection and what doesn't